OptimizelySDKCore/OptimizelySDKCore/NSArray+NSArray_OPTLY.m

```diff
@@ -0,0 +1,18 @@
+//
+//  NSArray+NSArray_OPTLY.m
+//  OptimizelySDKCore
+//
+//  Created by Abdur Rafay on 08/01/2018.
+//  Copyright © 2018 Optimizely. All rights reserved.
+//
+
+#import "NSArray+NSArray_OPTLY.h"
+
+@implementation NSArray (NSArray_OPTLY)
+
+- (BOOL)isEmpty {
+    BOOL isEmpty = self.count == 0;
+    return isEmpty;
+}
+
+@end
```

Comment:
My reaction to the NSArray+NSArray_OPTLY.m+.h category is very much
like my reaction to the NSString+NSString_OPTLY.m+h category which gets
my longer explanation below.

As an SDK, whenever (if ever) our Optimizely SDK creates a category on an
Apple class, we would have to prefix all our category method names with
something like "optly_" or "optimizely_" to prevent name collisions with potential
customer's own categories which may be defining these names.

Creating a category at all is inadvisable on an Apple class since this is a less
encapsulated approach.  I continue to recommend the private static "+"
approach of my previous comment about OPTLYMacros.h .

In case all we need is "self.count == 0" for an NSArray, then we should just
code "a.count == 0" in the OPTLYDecisionService.m code, not create a method
and category for this simple comparison.

In case we need something more like the OPTLYMacros.h macro
#define isEmptyArray(A)  ({ !(A && [A respondsToSelector:@selector(count)] && [(NSArray *)A count] > 0); })
you were proposing, then you could add a private static "+" "isEmptyArray:" method to OPTLYDecisionService.m .